2-mandatory/3-shopping-list.js

```diff
@@ -10,16 +10,18 @@
 
   The createShoppingList function should return an object with two properties:
     - "name" of the recipe, which is a string,
-    - "items", which is an arry of the missing ingredients that need to be on the shopping list
+    - "items", which is an array of the missing ingredients that need to be on the shopping list
 */
 
 let pantry = {
   fridgeContents: ["butter", "milk"],
   cupboardContents: ["salt", "tinned tomatoes", "oregano"],
 };
 
-function createShoppingList(recipe) {
-  // write code here
+let createShoppingList = (recipe) => {
+  let pantryContents = pantry.fridgeContents.concat(pantry.cupboardContents);
+  let missingIngredients = {} = recipe.ingredients.filter((ingredient) => !pantryContents.includes(ingredient));
+  return missingIngredients;
```

Comment:
You were very close here - you had the correct missing ingredients! If you look back at the instructions at the top of the file - you'll see why the test was failing. The test is expecting you to return an object with name and items properties.